ocs/agent/aggregator.py

```diff
@@ -18,6 +18,56 @@
     G3Module = object
 
 
+_g3_casts = {
+    str: core.G3String, int: core.G3Int, float: core.G3Double,
+}
+_g3_list_casts = {
+    str: core.G3VectorString, int: core.G3VectorInt, float: core.G3VectorDouble,
+}
+
+
+def g3_cast(data, time=False):
+    """
+    Casts a generic datatype into a corresponding G3 type. With:
+        int   -> G3Int
+        str   -> G3String
+        float -> G3Double
+    and lists of type X will go to G3VectorX. If ``time`` is set to True, will
+    convert to G3Time or G3VectorTime.
+
+    Args:
+        data (int, str, float, or list):
+            Generic data to be converted to a corresponding G3Type.
+        time (bool, optional):
+            If True, will try to cast to G3Time or G3VectorTime.
+
+    Returns:
+        g3_data:
+            Corresponding G3 datatype.
+    """
+    is_list = isinstance(data, list)
+    if is_list:
+        dtype = type(data[0])
+        if not all(isinstance(d, dtype) for d in data):
+            raise TypeError("Data list contains varying types!")
+    else:
+        dtype = type(data)
+    if dtype not in _g3_casts.keys():
+        raise TypeError("g3_cast does not support type {}. Type must"
+                        "be one of {}".format(dtype, _g3_casts.keys()))
+    if is_list:
+        if time:
+            return core.G3VectorTime(list(map(core.G3Time, data)))
+        else:
+            cast = _g3_list_casts[type(data[0])]
+            return cast(data)
+    else:
+        if time:
+            return core.G3Time(data)
```

Comment:
This won't convert from a unix timestamp to a G3Time... need to scale by G3Units.seconds, or something like that.